Source/Dafny/DafnyAst.cs

```diff
@@ -755,9 +755,12 @@ public abstract class Type {
           if (!rtd.AsTopLevelDecl.IsVisibleInScope(scope)) {
             // This can only mean "rtd" is a class/trait that is only provided, not revealed. For a provided class/trait,
             // it is the non-null type declaration that is visible, not the class/trait declaration itself.
+            // rtd can also be an OpaqueTypeDecl
```

Comment:
To confirm this belief, I suggest adding an `else` branch to the `if` below, asserting that `rtd` is an `OpaqueTypeDecl`.

Reply:
Done